cpp/src/arrow/util/variant.h

```diff
@@ -144,11 +144,11 @@ struct VariantImpl<Variant<T...>> : VariantStorage<T...> {
   void copy_to(...) const {}
 
   template <typename R, typename Visitor>
-  [[noreturn]] R visit_const(Visitor&& visitor) const {
+  [[noreturn]] R visit_const(Visitor&&) const {
     std::terminate();
   }
   template <typename R, typename Visitor>
-  [[noreturn]] R visit_mutable(Visitor&& visitor) {
+  [[noreturn]] R visit_mutable(Visitor&&) {
```

Comment:
Actually, I think there are other places where function arguments are not used, so this "issue" in itself could be resolved in its own JIRA issue.

Reply:
msvc complains only about these for now.